packages/graphql/src/schema-model/argument/model-adapters/ArgumentAdapter.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { AttributeType } from "../../attribute/AttributeType";
+import {
+    EnumType,
+    GraphQLBuiltInScalarType,
+    InterfaceType,
+    ListType,
+    Neo4jCartesianPointType,
+    Neo4jGraphQLNumberType,
+    Neo4jGraphQLSpatialType,
+    Neo4jGraphQLTemporalType,
+    Neo4jPointType,
+    ObjectType,
+    ScalarType,
+    UnionType,
+    UserScalarType,
+} from "../../attribute/AttributeType";
+import type { Argument } from "../Argument";
+
+export class ArgumentAdapter {
+    public name: string;
+    public type: AttributeType;
+    public description: string;
+    public defaultValue?: string;
+    private assertionOptions: {
+        includeLists: boolean;
+    };
+    constructor(argument: Argument) {
+        this.name = argument.name;
+        this.type = argument.type;
+        this.defaultValue = argument.defaultValue;
+        this.description = argument.description;
+        this.assertionOptions = {
+            includeLists: true,
+        };
+    }
+
+    /**
+     * Just a helper to get the wrapped type in case of a list, useful for the assertions
+     */
+    private getTypeForAssertion(includeLists: boolean) {
+        if (includeLists) {
+            return this.isList() ? this.type.ofType : this.type;
+        }
+        return this.type;
+    }
+
+    isBoolean(options = this.assertionOptions): boolean {
```

Comment:
Not blocking:  This duplicated code, maybe can be solved with https://www.typescriptlang.org/docs/handbook/mixins.html with a type-helper class, or maybe we could move outside these helpers directly outside. But I would want to revisit the "adapter" idea later in maybe just using directly the Schema Model and having an external GraphQL view to deal with the presentation logic for the typesName and helpers function outside, but we can properly design this later!

Reply:
so is anything in the scope of this PR?

Reply:
No, in my opinion is not worth spending time now to clean up this code if we already agreed that is not the final solution.